windows-installer/auto-update.ps1

```diff
@@ -0,0 +1,139 @@
+$_
+Assuming Certbot is not up-to-date.
+"@ "Warning"
+        $currentVersion = [System.Version]"0.0.0"
+    }
+
+    # Get latest remote certbot version
+    try {
+        [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+        $result = Invoke-RestMethod -Uri $certbotUpgradeApiURL
+        $latestVersion = $result.tag_name -replace '^v(\d+\.\d+\.\d+).*$', '$1'
+        $latestVersion = [System.Version]"$latestVersion"
+    } catch {
+        Throw-Error @"
+Could not get the latest remote certbot version. Error was:
+$_
+Aborting auto-upgrade process.
+"@
+    }
+
+    if ($currentVersion -ge $latestVersion) {
+        Write-Message "No upgrade is needed, Certbot is already at the latest version ($currentVersion)."
+    } else {
+        # Search for the Windows installer asset
+        $installerUrl = $null
+        foreach ($asset in $result.assets) {
+            if ($asset.name -match '^certbot-.*installer-win32\.exe$') {
+                $installerUrl = $asset.browser_download_url
+            }
+        }
+
+        if ($null -eq $installerUrl) {
+            Throw-Error "Could not find the URL for the latest Certbot for Windows installer."
+        }
+
+        Write-Message "Starting Certbot auto-upgrade from $currentVersion to $latestVersion ..."
+
+        $installerPath = "$env:TMP/certbot-installer-win32.exe"
+        try {
+            # Download the installer
+            Write-Message "Downloading the installer ..."
+            $webClient = New-Object System.Net.WebClient
+            $webClient.DownloadFile($installerUrl, $installerPath)
+
+            # Check installer has a valid signature from the Certbot release team
+            $signature = Get-AuthenticodeSignature $installerPath
+
+            if ($signature.Status -ne 'Valid') {
+                throw "Downloaded installer has no or invalid Authenticode signature."
+            }
+            $publicKey = $certbotSigningPubKey -replace '-+.*-+' -replace "`n" -replace "`r"
+            $refBinaryPublicKey = [System.Convert]::FromBase64String($publicKey)
+            $curBinaryPublicKey = $signature.SignerCertificate.PublicKey.EncodedKeyValue.RawData
+            $diff = Compare-Object -ReferenceObject $refBinaryPublicKey -DifferenceObject $curBinaryPublicKey
+            if ($diff) {
+                throw "Downloaded installer has not been signed by Certbot development team."
+            }
+
+            if (Test-Path $installDir\uninstall.exe) {
+                # Uninstall old Certbot first
+                Write-Message "Running the uninstaller for old version (install dir: $installDir) ..."
+                Start-Process -FilePath $installDir\uninstall.exe -ArgumentList "/S" -Wait
+            }
+            # Install new version of Certbot
+            Write-Message "Running the installer for new version (install dir: $installDir) ..."
+            Start-Process -FilePath $installerPath -ArgumentList "/S /D=$installDir" -Wait
```

Comment:
Invoking `/S` make the installer silent, and is specifically designed for non interactive calls. `-NoNewWindow` is not necessay here because the installer will not output anything, no logs or no UI.

And you have the answer for the second question: there is no logs to get here. The only outcome will be the exit code, that will trigger or not a global exception in the `auto-update.ps1` script (which is logged).

Reply:
Noted!